server/src/testFixtures/java/io/crate/testing/DataTypeTesting.java

```diff
@@ -66,23 +67,17 @@
 import io.crate.types.ShortType;
 import io.crate.types.StringType;
 import io.crate.types.TimestampType;
+import io.crate.types.UndefinedType;
 
 public class DataTypeTesting {
 
-    public static final List<DataType<?>> ALL_TYPES_EXCEPT_ARRAYS = new ArrayList<>();
-
-    static {
-        ALL_TYPES_EXCEPT_ARRAYS.addAll(DataTypes.PRIMITIVE_TYPES);
-        ALL_TYPES_EXCEPT_ARRAYS.add(DataTypes.GEO_POINT);
-        ALL_TYPES_EXCEPT_ARRAYS.add(DataTypes.GEO_SHAPE);
-        // ALL_TYPES_EXCEPT_ARRAYS.add(DataTypes.INTERVAL); Member of DataTypes.STORAGE_UNSUPPORTED
-        ALL_TYPES_EXCEPT_ARRAYS.add(DataTypes.UNTYPED_OBJECT);
-
-        ALL_TYPES_EXCEPT_ARRAYS.removeIf(x -> x.storageSupport() == null);
-    }
+    public static final Set<DataType<?>> ALL_STORED_TYPES_EXCEPT_ARRAYS = DataTypes.TYPES_BY_NAME_OR_ALIAS
+        .values().stream()
+        .filter(x -> x.storageSupport() != null && x.id() != ArrayType.ID && x.id() != UndefinedType.ID)
+        .collect(Collectors.toSet());
```

Comment:
Added IP and BIT manually since those types had issues with =[] and realised that would be better to actually add all stored types (can detect other bugs as well since we use it in other places)